TFT/src/User/Menu/LED.c

```diff
@@ -0,0 +1,575 @@
+#endif
+};
+
+#ifdef KEYBOARD_ON_LEFT
+const GUI_RECT ledColorRect = {
+  3 * KB_WIDTH, 0 * KB_HEIGHT, 4 * KB_WIDTH, 1 * KB_HEIGHT};
+
+const GUI_RECT ledPageRect = {
+  2 * CTRL_WIDTH, 4 * CTRL_HEIGHT, 3 * CTRL_WIDTH, 5 * KB_HEIGHT};
+#else
+const GUI_RECT ledColorRect = {
+  0 * KB_WIDTH, 0 * KB_HEIGHT, 1 * KB_WIDTH, 1 * KB_HEIGHT};
+
+const GUI_RECT ledPageRect = {
+  0 * CTRL_WIDTH, 4 * CTRL_HEIGHT, 1 * CTRL_WIDTH, 5 * KB_HEIGHT};
+#endif
+
+// area rectangles
+const GUI_RECT ledAreaRect[2] = {
+  {0, 0,               LCD_WIDTH, LCD_HEIGHT - CTRL_HEIGHT},  // keyboard area
+  {0, 4 * CTRL_HEIGHT, LCD_WIDTH, LCD_HEIGHT}                 // control bar area
+};
+
+const char *const ledKeyString[2] = {
+  "<",  // PREV
+  ">",  // NEXT
+};
+
+const char * const ledString[LED_VECT_SIZE] = {"R", "G", "B", "W", "P", "I"};
+
+const LED_VECT ledRed =   {0xFF, 0x00, 0x00, 0xFF, 0x00, 0x00};
+const LED_VECT ledGreen = {0x00, 0xFF, 0x00, 0xFF, 0x00, 0x00};
+const LED_VECT ledBlue =  {0x00, 0x00, 0xFF, 0xFF, 0x00, 0x00};
+const LED_VECT ledWhite = {0xFF, 0xFF, 0xFF, 0xFF, 0x00, 0x00};
+const LED_VECT ledOff =   {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+LED_VECT ledValue = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+uint8_t ledPage = 0;
+uint8_t ledIndex = 0;
+
+void ledSendValue(const LED_VECT * led)
+{
+  storeCmd("M150 R%d U%d B%d W%d P%d I%d\n", (*led)[0], (*led)[1], (*led)[2], (*led)[3], (*led)[4], (*led)[5]);
+}
+
+void ledGetValue(LED_VECT * led)
+{
+  for (int i = 0; i < LED_VECT_SIZE; i++)
+    (*led)[i] = ledValue[i];
+}
+
+void ledSetValue(const LED_VECT * led)
+{
+  for (int i = 0; i < LED_VECT_SIZE; i++)
+    ledValue[i] = (*led)[i];
+}
+
+uint8_t ledGetComponetIndex(uint8_t index)
+{
+  return ledPage * PAGE_ITEMS + index;
+}
+
+uint8_t ledGetComponentValue(uint8_t index)
+{
+  return ledValue[ledGetComponetIndex(index)];
+}
+
+static inline uint8_t ledEditComponentValue(uint8_t index)
+{
+  uint8_t realIndex = ledGetComponetIndex(index);
+
+  return ledValue[realIndex] = editIntValue(LED_MIN_VALUE, LED_MAX_VALUE, ledValue[realIndex], ledValue[realIndex]);
+}
+
+uint8_t ledUpdateComponentValue(uint8_t index, int8_t unit, int8_t direction)
+{
+  uint8_t realIndex = ledGetComponetIndex(index);
+
+  return ledValue[realIndex] = NOBEYOND(LED_MIN_VALUE, ledValue[realIndex] + (int16_t) (direction * unit), LED_MAX_VALUE);
+}
+
+uint8_t ledGetControlIndex(uint8_t keyNum)
+{
+  return (keyNum - (LED_KEY_OK + 1)) / 4;
+}
+
+uint8_t ledGetControlSubIndex(uint8_t keyNum)
+{
+  return (keyNum - (LED_KEY_OK + 1)) % 4;
+}
+
+uint16_t ledGetRGBColor(LED_VECT * led)
+{
+  return (((*led)[0] & 0x001F) << 11) | (((*led)[1] & 0x003F) << 5) | (((*led)[2] & 0x001F));  // RGB color in RGB 565 16 bit format
+}
+
+uint16_t ledGetComponentRGBColor(uint8_t component, uint8_t index)
+{
+  LED_VECT led = {0x00, 0x00, 0x00, 0x00, 0x00, 0x00};  // component RGB color in RGB 565 16 bit format
+
+  led[index] = component;
```

Comment:
is it necessary to add space between just two lines?

Reply:
usually I always separate variable definition